src/bitcoin-cli.cpp

```diff
@@ -1059,7 +1059,10 @@ static void ParseGetInfoResult(UniValue& result)
         result_string += "\n";
     }
 
-    result_string += strprintf("%sWarnings:%s %s", YELLOW, RESET, result["warnings"].getValStr());
+    if (result["warnings"].getValStr()!="")
+    {
+        result_string += strprintf("%sWarnings:%s %s", YELLOW, RESET, result["warnings"].getValStr());
+    }
```

Comment:
suggestions

- don't add extra line breaks when no warning is present
- avoid running getValStr() twice
- use empty()
- clang-format of brackets in the conditional

```diff
--- a/src/bitcoin-cli.cpp
+++ b/src/bitcoin-cli.cpp
@@ -1038,7 +1038,7 @@ static void ParseGetInfoResult(UniValue& result)
         result_string += strprintf("Transaction fee rate (-paytxfee) (%s/kvB): %s\n\n", CURRENCY_UNIT, result["paytxfee"].getValStr());
     }
     if (!result["balance"].isNull()) {
-        result_string += strprintf("%sBalance:%s %s\n\n", CYAN, RESET, result["balance"].getValStr());
+        result_string += strprintf("%sBalance:%s %s\n", CYAN, RESET, result["balance"].getValStr());
     }
 
     if (!result["balances"].isNull()) {
@@ -1056,13 +1056,13 @@ static void ParseGetInfoResult(UniValue& result)
                                        result["balances"][wallet].getValStr(),
                                        wallet.empty() ? "\"\"" : wallet);
         }
-        result_string += "\n";
     }
 
-    if (result["warnings"].getValStr()!="")
-    {
-        result_string += strprintf("%sWarnings:%s %s", YELLOW, RESET, result["warnings"].getValStr());
+    const std::string warnings{result["warnings"].getValStr()};
+    if (!warnings.empty()) {
+        result_string += strprintf("\n%sWarnings:%s %s", YELLOW, RESET, warnings);
     }
+
     result.setStr(result_string);
 }
```
With this suggestion, I think there is still an extra line break after the balance/balances, if you can look into it.

Reply:
@jonatack thanks for review and suggestion. Made changes, tested and it looks good. Let me know if there are any issues with line breaks:

With warning: https://pastebin.com/raw/eN3SGjiX
Without warning: https://pastebin.com/raw/frfnqfvU

Reply:
As I mentioned, there is still an extra line, and the commits should be squashed, as the extra line is created by the changes here.

Reply:
Done. Removed extra lines: https://pastebin.com/raw/bZk4YNj5